pmm/pmm2-release.groovy

```diff
@@ -549,40 +544,42 @@ ENDSSH
-
-                            if [ -n "$SHA" ] && [ -n "$REPO" ]; then
-                                rm -fr $PACKAGE || true
-                                mkdir $PACKAGE
-                                pushd $PACKAGE >/dev/null
-                                    git clone https://github.com/$REPO ./
-                                    # The default is https, so we want to set it to ssh
-                                    git remote set-url origin git@github.com:$REPO.git
-                                    git checkout $SHA
-                                    echo "SHA: $(git rev-parse HEAD)"
-
-                                    git tag --message="Version $TAG." --sign $TAG
-
-                                    # If the tag already exists, we want to delete it and re-tag this SHA
-                                    if [ $? -eq 128 ]; then
-                                        git tag --delete $TAG
-                                        git push --delete origin $TAG
-                                        git tag --message="Version $TAG." --sign $TAG
-                                    fi
-
-                                    if [ $? -eq 0 ]; then
-                                        git push origin $TAG
-                                    else
-                                        echo "Error: $?"
-                                    fi
-                                popd >/dev/null
-                            else
-                                echo "Warning: the repository $REPO won't get tagged with ${VERSION}"
-                            fi
+                            rm -fr $PACKAGE || true
+                            mkdir -p $PACKAGE
+
+                            pushd $PACKAGE >/dev/null
+                                REPO=${repos["$PACKAGE"]}
+                                git clone https://github.com/$REPO ./
+                                # The default is https, and we want to set it to ssh
+                                git remote set-url origin git@github.com:$REPO.git
+                                
+                                BRANCH="pmm-${VERSION}"
+                                git checkout "$BRANCH"
+                                if [ $? -ne 0 ]; then
+                                  continue
+                                fi
+                                echo "SHA: $(git rev-parse HEAD)"
+                                echo "Branch: $(git branch --show-current)"
+
+                                git tag --message="Version $TAG." --sign "$TAG"
+
+                                # If the tag already exists, we want to delete it and re-tag this SHA
+                                if [ $? -eq 128 ]; then
+                                    git tag --delete "$TAG"
+                                    git push --delete origin "$TAG"
+                                    git tag --message="Version $TAG." --sign "$TAG"
+                                fi
```

Comment:
maybe we can check if tag exists before adding it?
```
is_set=$(git tag | grep $TAG)
if [ $is_set = 1 ]; then
    git tag --delete "$TAG"
    git push --delete origin "$TAG"
fi
git tag --message="Version $TAG." --sign "$TAG"
```

Reply:
That's a good one, will do!